README.md

```diff
@@ -41,6 +39,45 @@ asdf uses a `.tool-versions` file for auto-switching between software versions.
 legacy_version_file = yes
 ```
 
+## Partial and codename versions
+
+Many version managers allow you to use partial versions (e.g. `v10`) or NodeJS codenames (e.g. `lts/hydrogen`) in version files, which are resolved at runtime. However, this can be risky as it is not guaranteed that all developers will use the same version, leading to non-reproducibility. In `asdf`, we prioritize reproducibility, so you cannot use partial versions or codenames in a `.tool-versions` file.
+
+To address this, we offer an escape hatch for legacy version files. If you are comfortable with non-reproducibility issues, you can provide your own resolution script as an environment variable `ASDF_NODEJS_RESOLVE_VERSION_CMD`. You can export this variable from your shell rc file and it will become the default behavior.
+
+It is important to be aware of the risks of non-reproducibility. Debugging can become more challenging and bugs may leak into production if the deployed node version differs from the one used in development. Ideally, maintainers should be encouraged to pin the version to a specific release to avoid these issues.
+
+If non-reproducibility is not a concern for you, you can use one of the following resolve scripts in your shell rc file:
+
+```bash
+# This script will always resolve to the latest version of said partial, even if not installed.
+# If a new version is released for that partial/codename, asdf will prompt you to install it.
+export ASDF_NODEJS_RESOLVE_VERSION_CMD='asdf nodejs latest "$1"'
+
+# This script will resolve to the latest *installed* version that satisfies the partial/codename.
+# If new versions that match said partial are released, you need to install them with `asdf install nodejs <version>`.
+# The newly installed version will be picked up automatically.
+export ASDF_NODEJS_RESOLVE_VERSION_CMD='asdf nodejs latest "$1" --installed'
+```
+
+These are just examples, and you can provide your own script that resolves to your preferred version.
```

Comment:
Do you have any other resolver functions in mind? If not, I'm wondering if we could simplify this interface so something simpler:

```bash
# For latest available version matching
export ASDF_NODEJS_RESOLVE_VERSION=latest_available
# For latest installed version matching
export ASDF_NODEJS_RESOLVE_VERSION=latest_installed
```

Reply:
I was thinking about making the barrier more explicit so people could spend some thought on wether dynamic versions are worth it. But I agree that having the algorithms more explicit would make a better UI. I will change it to offer only these two strategies for now